packages/tools/src/stateManagement/annotation/annotationState.ts

```diff
@@ -1,151 +1,146 @@
 import {
-  AnnotationAddedEventDetail,
-  AnnotationRemovedEventDetail,
-} from '../../types/EventTypes';
+  triggerAnnotationAddedForElement,
+  triggerAnnotationAddedForFOR,
+} from './helpers/state';
+
+// our default annotation manager
+let defaultManager = defaultFrameOfReferenceSpecificAnnotationManager;
 
 /**
  * It returns the default annotations manager.
  * @returns the singleton default annotations manager.
  */
-function getDefaultAnnotationManager() {
-  return defaultFrameOfReferenceSpecificAnnotationManager;
+function getAnnotationManager() {
+  return defaultManager;
 }
 
 /**
- * Given an element, return the FrameOfReferenceSpecificStateManager for that
- * element
- * @param element - The element that the state manager is managing the state of.
- * @returns The default state manager
+ * Set the annotation manager to be used for rendering, adding, removing, etc.
+ * @param annotationManager - The annotation manager to be used
  */
-function getViewportSpecificAnnotationManager(
-  element?: Types.IEnabledElement | HTMLDivElement
-) {
-  // TODO:
-  // We may want multiple FrameOfReferenceSpecificStateManagers.
-  // E.g. displaying two different radiologists annotations on the same underlying data/FoR.
-
-  // Just return the default for now.
+function setAnnotationManager(annotationManager) {
+  defaultManager = annotationManager;
+}
 
-  return defaultFrameOfReferenceSpecificAnnotationManager;
+// set back to default frameOfReferenceSpecificAnnotationManager
+function resetAnnotationManager() {
+  defaultManager = defaultFrameOfReferenceSpecificAnnotationManager;
 }
 
 /**
- * Returns the annotations for the `FrameOfReference` of the `Viewport`
- * being viewed by the cornerstone3D enabled `element`.
+ * Returns the annotations for a given tool with the provided options that is
+ * used to filter annotations based on the annotation manager.
+ *
+ * In our default implementation, the options are the element and/or the FrameOfReferenceUID.
+ * Hence, the getAnnotations function will return the annotations for the given tool
+ * that are associated with the FrameOfReferenceUID.
  *
- * @param element - The HTML element.
  * @param toolName - The name of the tool.
+ * @param options - The options to filter the annotations (element and/or FrameOfReferenceUID by default)
  * @returns The annotations corresponding to the Frame of Reference and the toolName.
  */
 function getAnnotations(
-  element: HTMLDivElement,
-  toolName: string
+  toolName: string,
+  options: Record<string, any>
 ): Annotations {
-  const enabledElement = getEnabledElement(element);
-  const annotationManager =
-    getViewportSpecificAnnotationManager(enabledElement);
-  const { FrameOfReferenceUID } = enabledElement;
-
-  return annotationManager.get(FrameOfReferenceUID, toolName);
+  const manager = getAnnotationManager();
+  const groupKey = manager.getGroupKey(options);
+  return manager.getAnnotations(groupKey, toolName) as Annotations;
```

Comment:
It seems here that the output of type of `manager.getAnnotations` could be `undefined`, but is being coerced into `Annotations`. Would it make sense to return an empty Array if annotations are undefined, to respect to return type?

Reply:
Good catch, thanks!